tests/test_checkers.py

```diff
@@ -0,0 +1,318 @@
+"""Tests for the loading, verification, and use of plugin-based checkers."""
+
+
+from gator import arguments
+from gator import checkers
+
+import os
+import pytest
+import sys
+
+from unittest.mock import patch
+
+
+@pytest.fixture(autouse=True)
+def reset_checker_source():
+    """Before running any test in this suite always reset the source of the checkers in pluginbase."""
+    # note that performing this reset ensures test independence and
+    # avoids test flakiness for single-test runs or different test orderings
+    checkers.reset_source()
+
+
+def test_reset_checker_source():
+    """Ensure that reset of the checker source works correctly."""
+    assert checkers.CHECKER_SOURCE is None
+    checkers.reset_source()
```

Comment:
This doesn't seem to test the resetting... if it was `None` before nothing could happen and it would still pass: a plugin should be loaded or some other action done to be reset here.

Reply:
Thanks for this suggestion! I added a call to the new `load_checker` function so that a full check is loaded, then the `reset_source` is called and then the assertion is run. This improved test is passing.